File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int count = 0;
        vector<int> nested;
  
        for(int i = 0; i < s.length(); i++){
            if(s[i] == '('){
                count++;
                nested.push_back(count);
            }
            if(s[i] == ')'){
                count--;
            }
        }
        if (nested.empty()) {
            return 0;
        }
        int high = *max_element(nested.begin(),nested.end());
        return high;
    }
};

File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        count = 0
        freq = []

        for i in s:
            if i == "(":
                count += 1
                freq.append(count)
            
            if i == ")":
                count -= 1

        maxEle = float('-inf')

        if len(freq) == 0:
            return 0

        for i in freq:
            if i > maxEle:
                maxEle = i

        return maxEle
